Userland/Libraries/LibJS/Heap/Heap.cpp

```diff
@@ -145,6 +145,8 @@ __attribute__((no_sanitize("address"))) void Heap::gather_conservative_roots(Has
     for (FlatPtr stack_address = stack_reference; stack_address < stack_info.top(); stack_address += sizeof(FlatPtr)) {
         auto data = *reinterpret_cast<FlatPtr*>(stack_address);
         possible_pointers.set(data);
+        if ((data & SHIFTED_IS_CELL_PATTERN) == SHIFTED_IS_CELL_PATTERN)
```

Comment:
If the possible pointer matches the shift cell pattern it can't be a valid canonical pointer, so there's no point in adding its raw contents to the possible list (aka move the .set() above to the else case)
Also, shifted_is_cell_pattern is a u64 right? But data is a FlatPtr which is a u32 on i686, so how does this work?

Reply:
Yeah moving that to else is a good point.

For i686 this does work because we just check the lower part of Value separately from the top part and never go into the if but I'll add a comment since it isn't obvious.